src/Authentication/Authenticators/Session.php

```diff
@@ -88,9 +93,59 @@ public function attempt(array $credentials): Result
 
         $this->recordLoginAttempt($credentials, true, $ipAddress, $userAgent, $user->getAuthId());
 
+        // If an action has been defined for login, start it up.
+        $actionClass = setting('Auth.actions')['login'] ?? null;
+        if (! empty($actionClass)) {
+            session()->set('auth_action', $actionClass);
+        } else {
+            // a successful login
+            Events::trigger('login', $user);
+        }
+
         return $result;
     }
 
+    /**
+     * Check token in Action
+     *
+     * @param string $type  Action type. 'email_2fa' or 'email_activate'
+     * @param string $token Token to check
+     */
+    public function checkAction(string $type, string $token): bool
+    {
+        $user = $this->loggedIn()
+            ? $this->getUser()
+            : null;
+
+        $identity = $user->getIdentity($type);
```

Comment:
If `$user` ever is actually `null` this will cause an error.

Reply:
Added null check.